packages/eas-cli/src/metadata/apple/tasks/app-review-detail.ts

```diff
@@ -0,0 +1,57 @@
+import { AppStoreReviewDetail } from '@expo/apple-utils';
+import assert from 'assert';
+import chalk from 'chalk';
+
+import Log from '../../../log';
+import { logAsync } from '../../utils/log';
+import { AppleTask, TaskDownloadOptions, TaskPrepareOptions, TaskUploadOptions } from '../task';
+
+export type AppReviewData = {
+  /** The current app info that should be edited */
+  reviewDetail: AppStoreReviewDetail;
+};
+
+export class AppReviewDetailTask extends AppleTask {
+  public name = (): string => 'app review detail';
+
+  public async prepareAsync({ context }: TaskPrepareOptions): Promise<void> {
+    assert(context.version, `App version not initialized, can't download store review details`);
+    context.reviewDetail = (await context.version.getAppStoreReviewDetailAsync()) ?? undefined;
+  }
+
+  public async downloadAsync({ config, context }: TaskDownloadOptions): Promise<void> {
+    if (context.reviewDetail) {
+      config.setReviewDetails(context.reviewDetail.attributes);
+    }
+  }
+
+  public async uploadAsync({ config, context }: TaskUploadOptions): Promise<void> {
+    const reviewDetail = config.getReviewDetails();
+    if (!reviewDetail) {
+      return Log.log(chalk`{dim - Skipped store review details, not configured}`);
+    }
+
```

Comment:
It's almost never happening. The task has multiple implicit dependencies, and all are asserted. It could happen whenever the tasks are executed in the wrong order, or when the task that loaded this model failed for some reason.

Or when testing, and you forgot to pass the implicit dependency :)